### PR TITLE
Improve corefx test coverage for exception types

### PR DESCRIPTION
**Pull request by jkotas**
Contributes to dotnet/coreclr#12782

### CODE REVIEW

File: src/System.Runtime/tests/System/ArgumentExceptionTests.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+using ExceptionUtility = System.Tests.ExceptionUtility;
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/ArgumentExceptionTests.cs]: Why is this necessary?

File: src/System.Runtime/tests/System/IO/PathTooLongException.InteropTests.cs [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 using System.Runtime.InteropServices;
 using Xunit;
+using ExceptionUtility = System.Tests.ExceptionUtility;
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/IO/PathTooLongException.InteropTests.cs]: Could we just do `using System.Tests;`?

File: src/System.Runtime/tests/System/ObjectDisposedExceptionTests.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+{
+    public static class ObjectDisposedExceptionTests
+    {
+        private const int COR_E_OBJECTDISPOSED = unchecked((int)0x80131622);
+
+        [Fact]
+        public static void Ctor_String()
+        {
+            string objectName = "theObject";
+            var exception = new ObjectDisposedException(objectName);
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/ObjectDisposedExceptionTests.cs]: Nit: it'd be nice to also validate that null is accepted

File: src/System.Runtime/tests/System/ArgumentExceptionTests.cs (hunk 1 above)
Comment (jkotas) [reply src/System.Runtime/tests/System/ArgumentExceptionTests.cs]: I have used the pattern used in the existing tests. I can change it to `using System.Tests`

**Comment by jkotas** [comment]
```
Failed obtain Helix work item information:
[Pipeline] echo
java.lang.NullPointerException
[Pipeline] echo
[]
[Pipeline] }
Will try again after 3 min 0 sec
Cancelling nested steps due to timeout
```

**Comment by jkotas** [comment]
@dotnet-bot test Linux x64 Release Build  please

File: src/System.Runtime/tests/System/ArgumentExceptionTests.cs [hunk 4]
```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+using System.Tests;
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/ArgumentExceptionTests.cs]: These `using System.Tests;`s shouldn't be necessary when all of the tests are in `namespace System.Tests;`.

**Comment by danmosemsft** [comment]
@pjanotti I know work is not completed, but do we have a recent code coverage report that includes corelib? That may help Jan see what is already covered by corefx tests.

**Comment by jkotas** [comment]
@pjanotti I have tried to cross-reference the CoreLib coverage in the report with the actual tests and it does not make sense to me. It looks like many tests that are exercising types in CoreLib are not actually included in the report. Do I need to add `<TestRuntime>true</TestRuntime>` tag on all tests that are targeted at testing CoreLib? (e.g. this tag is missing on System.Runtime tests that are the main tests testing CoreLib).

**Comment by pjanotti** [comment]
@jkotas yes, the tags are required. I will start a PR for that. I didn't try recently but you should be able to also get it from the command line by adding `/p:CodeCoverageAssemblies:System.Private.CoreLib` (I will give a try to it later).

**Comment by pjanotti** [comment]
`/p:CodeCoverageAssemblies:System.Private.CoreLib` works, I will take a little longer with the PR since I want to do another scrub since I clearly missed some projects on previous commit.

**Comment by jkotas** [comment]
@pjanotti I have created https://github.com/dotnet/corefx/pull/32700 to fixup the TestRuntime tags

**Comment by pjanotti** [comment]
Thanks @jkotas per review comment I'll add a related one to buildtools but it doesn't need to wait for it.
